string method
---
bool result
string message